crates/core/src/otel.rs

```diff
@@ -23,6 +30,86 @@ pub struct OtelConfig {
     pub metrics_endpoint: Option<String>,
     /// Overrides the OpenTelemetry endpoint for logs.
     pub logs_endpoint: Option<String>,
+    /// Determines whether grpc or http will be used for exporting the telemetry.
+    pub protocol: Option<String>,
```

Comment:
If you make this an `Option<OtelProtocol>` it would definitely simplify other parts of the code 😄 What I would want to double check though is that it serializes cleanly to JSON as the lowercase variant

Reply:
That being said, you may as well just make it non-optional, since you _have_ to have a protocol right?

Reply:
I made this non-optional and added a `impl Default for OtelProtocol` which takes care of the case where a None would be passed via the command-line.